shotover-proxy/src/server.rs

```diff
@@ -461,6 +478,21 @@ impl<C: Codec + 'static> Handler<C> {
 
             debug!("client details: {:?}", &self.client_details);
 
+            if let Some(limiter) = limiter.as_ref() {
+                messages = messages
+                    .into_iter()
```

Comment:
No need to recreate the vec we should be able to just `for message in &mut messages`